app/lib/shared/configuration.dart

```diff
@@ -130,12 +142,47 @@ class Configuration {
   /// The identifier of admins.
   final List<AdminId> admins;
 
+  @override
+  List<Object> get props => [
+        projectId,
+        packageBucketName,
+        dartdocStorageBucketName,
+        popularityDumpBucketName,
+        searchSnapshotBucketName,
+        backupSnapshotBucketName,
+        searchServicePrefix,
+        storageBaseUrl,
+        pubClientAudience,
+        pubSiteAudience,
+        adminAudience,
+        gmailRelayServiceAccount,
+        gmailRelayImpersonatedGSuiteUser,
+        uploadSignerServiceAccount,
+        blockRobots,
+        productionHosts,
+        primaryApiUri,
+        primarySiteUri,
+        admins,
+      ];
+
+  factory Configuration.fromJson(Map<String, dynamic> json) =>
+      _$ConfigurationFromJson(json);
+  Map<String, dynamic> toJson() => _$ConfigurationToJson(this);
+
+  factory Configuration.fromYamlFile(final String path) {
+    final file = File(path);
+    final content = file.readAsStringSync();
+    final map =
+        json.decode(json.encode(loadYaml(content))) as Map<String, dynamic>;
+    return Configuration.fromJson(map);
+  }
+
   /// Create a configuration for production deployment.
   ///
   /// This will use the Datastore from the cloud project and the Cloud Storage
   /// bucket 'pub-packages'. The credentials for accessing the Cloud
   /// Storage is retrieved from the Datastore.
-  factory Configuration._prod() {
+  factory Configuration.prodConfig() {
```

Comment:
the method name was changed to simplify access from tests.

Reply:
Let's add @visibleForTesting annotation to it.